HOWTO_android.md

```diff
@@ -0,0 +1,637 @@
+
+### 1. Pull down the GfxReconstruct source
+
+If you haven't already, pull down the GfxReconstruct source from GitHub so that
+it also updates any necessary submodules as part of the clone operation:
+
+```bash
+git clone --recurse-submodules https://github.com/LunarG/gfxreconstruct.git
+```
+
+The full location of the `gfxreconstruct` folder generated from this step
+is to be used in all cases below for `{gfxreconstruct_root}`.
+
+### 2. Pull down the source for Vulkan Samples
+
+```bash
+git clone --recurse-submodules https://github.com/KhronosGroup/Vulkan-Samples.git
+```
+
+### 3. Enter the Vulkan Samples source directory
+
+```bash
+cd Vulkan-Samples
+```
+
+### 4. Build the gradle files
+
+```bash
+./bldsys/scripts/generate_android_gradle.sh
+```
+
+### 5. Edit the top-level settings.gradle file
+   (./build/android_gradle/settings.gradle)
+   and add the following lines to the end
+
+```text
+include(':app',':VkLayer_gfxreconstruct')
+project(':VkLayer_gfxreconstruct').projectDir = file('{gfxreconstruct_root}/android/layer')
+```
+
+**NOTE** Replacing {gfxreconstruct_root} with the location of the source you
+cloned from the GfxReconstruct repo in step 1.
+
+### 6. Add the capture layer to the samples application
+
+Open ./build/android_gradle/app/build.gradle and add the following line at the
+top of the `dependencies` block at the end of the file:
+
+```text
+    implementation project(':VkLayer_gfxreconstruct')
+```
+
+It should look something like this when you are done:
+
+```text
+dependencies {
+    implementation project(':VkLayer_gfxreconstruct')
+	implementation 'androidx.appcompat:appcompat:1.3.0'
+	implementation 'com.google.android.material:material:1.4.0'
+	implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+}
+```
+
+### 7. Change to the appropriate build folder
+
+From the top-level samples folder you now need to enter the following sub-
+folder to build
+
+```bash
+cd build/android_gradle
+```
+
+### 8. Build the appropriate target
+
+Some common targets are `assembleDebug` and `assembleRelease`.
+We'll build for debug in this scenario:
+
+```bash
+gradle assembleDebug
+```
+
+
+## Capturing Vulkan Samples
+
+This example follows using the Khronos Vulkan Samples built above and assumes
+that you built the debug version.
+The instructions really only work on non-rooted phones if you manually build and
+install the application with support for the GfxReconstruct layer.
+The benefit of building the application in this way is that it will
+automatically incorporate the GfxReconstruct capture layer into the .APK file
+for the built application.
+
+### 1. Remove any previous version of the Vulkan Samples app
+
+We'll intentionally remove any previous instance of the application we're going
+to trace.
+In this case, the full name of the Vulkan samples app for Android is called
+"com.khronos.vulkan_samples":
+
+```bash
+adb uninstall com.khronos.vulkan_samples
```

Comment:
I don't remember having to uninstall.  Doesn't install overwrite the old app?

Reply:
No.  Install can actually fail sometimes if the current application is doing something.  The uninstall process forces even stubborn apps to halt, be removed, and leave a clean slate.